inference-engine/src/vpu/common/src/ngraph/transformations/convert_extract_image_patches_to_reorg_yolo_vpu.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "vpu/ngraph/transformations/convert_extract_image_patches_to_reorg_yolo_vpu.hpp"
+
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace vpu {
+
+ConvertExtractImagePatchesToReorgYoloVPU::ConvertExtractImagePatchesToReorgYoloVPU() {
+    const auto image = std::make_shared<ngraph::pattern::op::Label>(ngraph::element::f32, ngraph::Shape{1, 1, 1, 1});
+    const auto eip = std::make_shared<ngraph::opset5::ExtractImagePatches>(
+            image, ngraph::Shape{1, 1}, ngraph::Strides{1, 1}, ngraph::Shape{1, 1}, ngraph::op::PadType::VALID);
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        const auto extractImagePatches =  std::dynamic_pointer_cast<ngraph::opset5::ExtractImagePatches>(m.get_match_root());
+
+        /*
+         * In this transformation we replace ExtractImagePatches operation to ReorgYolo operation
+         * if ExtractImagePatches operation attributes obey the following conditions:
+         *
+         * EIP.sizes = EIP.strides
+         * EIP.rates = {1, 1}
+         * Spatial dimensions of input tensor must be divisible by EIP.strides
+         *
+         */
+
+        if (!extractImagePatches || m_transformation_callback(extractImagePatches)) {
+            return false;
+        }
+
+
+        if (extractImagePatches->get_strides() != extractImagePatches->get_sizes()) {
+            return false;
+        }
+
+        const auto& inputPartialShape = extractImagePatches->get_input_partial_shape(0);
+        const auto& sizes = extractImagePatches->get_sizes();
+        const auto& strides = extractImagePatches->get_strides();
+        const auto& rates = extractImagePatches->get_rates();
+
+        // Check that ExtractImagePatches input have static shape and rank == 4
```

Comment:
I think this comment and comments below are excessive, it's very clear from the code what they are actually checking.

Reply:
Done